extensions/2.0/Khronos/KHR_lights_environment/README.md

```diff
@@ -0,0 +1,196 @@
+            }
+        ]
+    }
+}
+```
+
+## Specular Radiance Cubemaps
+
+The cubemap used for specular radiance is defined as a cubemap containing separate images for each cube face. 
+The mip levels shall evenly map to roughness values from 0 to 1 in the PBR material and should be generated with a principled multi-scatter GGX normal distribution. The data in the maps represents illuminance in candela per square meter (nit).
+
+If mip-levels are not included, then the entire mip chain of images should not be generated.  
+Instead, the lowest-resolution mip should have sufficient size to represent the maximally-blurred radiance map corresponding to roughness=1.  
+The texture references defines the largest dimension of mip 0 and should give the loading runtime the information it needs to generate the remainder of the mip chain and sample the appropriate mip level in the shader.  
+
+Cube faces are defined in the KTX2 format specification, users of this extension must follow the KTX2 cubemap specification.  
+
+## Implementation Note 
+This section is non-normative  
+
+The cubemap textures are used both as direct light contribution and as source for reflection.  
+A performant way of achieving this is to store the reflection at different roughness values in mip-levels,   
+where mip-level 0 is for a mirror like reflection (roughness = 0) and the last mip-level is for maximum roughness (roughness = 0).  
+It is recommended to map roughness = 1 to a mip-level size that is larger than 1 * 1 pixels to avoid blockiness, it is generally enough to stop at 16 * 16 pixels.  
+To avoid oversampling of flat surfaces (low roughness values) it is recommended to use a 3D cubemap when possible.  
+
+[Runtime filtering of mip-levels](https://developer.nvidia.com/gpugems/gpugems3/part-iii-rendering/chapter-20-gpu-based-importance-sampling)
+
+
+## Irradiance Coefficients
+
+This extension may use spherical harmonic coefficients to define irradiance used for diffuse lighting.  
+Coefficients are calculated for the first 3 SH bands (l=2) and take the form of a 9x3 array.  
+
+## Implementation Note
+This section is non-normative  
+
+Implementations may calculate the irradiance from the specular radiance cubemap and use instead.  
+One possible benefit with spherical harmonics is that it is generally enough evaluate the harmonics on a per vertex basis, instead of sampling a cubemap on a per fragment basis.  
+
+[Realtime Image Based Lighting using Spherical Harmonics](https://metashapes.com/blog/realtime-image-based-lighting-using-spherical-harmonics/)  
+[An Efficient Representation for Irradiance Environment Maps](http://graphics.stanford.edu/papers/envmap/)  
+
+### Using the environment light
+
+The environment light is utilized by a scene.  
+Each scene can have a single environment light attached to it by defining the `extensions.KHR_lights_environment` property and, within that, an index into the `lights` array using the `light` property.
```

Comment:
It may be desirable to also be able to add environment light references to materials as well, with the one on the scene here being a default fallback.

Reply:
Not sure as how to derive the use of another environment light based on a material property....
Could you please elaborate a bit more on the usecase for this?

Reply:
What I meant is objects using a given material would also be lit by a given environment map. This is how environment maps are currently specified in threejs.

The primary use case I am looking at for this is large environments where a single environment map will not be sufficient to cover the entire model. You can imagine an interior with multiple rooms, each of which might need their own environment map. There are definitely many ways to slice that and I dont think it has to be at the material level.